yoti-sdk-api/src/main/java/com/yoti/api/client/identity/policy/WantedAttribute.java

```diff
@@ -0,0 +1,145 @@
+            boolean optional,
+            Boolean acceptSelfAsserted,
+            List<Constraint> constraints) {
+        notNullOrEmpty(name, Property.NAME);
+
+        this.name = name;
+        this.derivation = derivation;
+        this.optional = optional;
+        this.acceptSelfAsserted = acceptSelfAsserted;
+
+        if (constraints == null) {
+            this.constraints = Collections.emptyList();
+        } else {
+            this.constraints = constraints;
+        }
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDerivation() {
+        return derivation;
+    }
+
+    public boolean isOptional() {
+        return optional;
+    }
+
+    public Boolean getAcceptSelfAsserted() {
+        return acceptSelfAsserted;
+    }
+
+    public List<Constraint> getConstraints() {
+        return constraints;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public String name() {
+        return name;
+    }
+
+    public String derivation() {
+        return derivation;
+    }
+
+    public List<Constraint> constraints() {
+        return constraints;
+    }
+
+    public static class Builder {
+
+        private String name;
+        private String derivation;
+        private boolean optional;
+        private Boolean acceptSelfAsserted;
+        private List<Constraint> constraints;
+
+        private Builder() {
+            this.constraints = new ArrayList<>();
+        }
+        
+        public Builder withName(String name) {
+            this.name = name;
+            return this;
+        }
+        
+        public Builder withDerivation(String derivation) {
+            this.derivation = derivation;
+            return this;
+        }
+        
+        public Builder withOptional(boolean optional) {
+            this.optional = optional;
+            return this;
+        }
+
+        public Builder withAcceptSelfAsserted(boolean acceptSelfAsserted) {
+            this.acceptSelfAsserted = acceptSelfAsserted;
+            return this;
+        }
+
+        public Builder withConstraints(List<Constraint> constraints) {
+            this.constraints = Collections.unmodifiableList(constraints);
+            return this;
+        }
+
+        public Builder withConstraint(Constraint constraint) {
+            this.constraints.add(constraint);
+            return this;
+        }
+
+        public WantedAttribute build() {
+            return new WantedAttribute(name, derivation, optional, acceptSelfAsserted, constraints);
+        }
+
+    }
+
+    private static final class Property {
+
+        private static final String NAME = "name";
+        private static final String DERIVATION = "derivation";
+        private static final String OPTIONAL = "optional";
+        private static final String ACCEPT_SELF_ASSERTED = "accept_self_asserted";
+        private static final String CONSTRAINTS = "constraints";
+
+        private Property() { }
```

Comment:
(nit) Since these `Property` classes are `private` ones, I think the ctor could be dropped - it's no like the outer class is going to be doing `new Property` (it's pretty obvious its a constants class).